### ISSUE TITLE
this javascript code's tip is misleading, this is not inaccessible all time

### ISSUE THREAD

**Issue by alexxxcs1** [microsoft/vscode#94109]
Version: 1.43.2
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.3.0

MacOS version: 10.15.3

List of extensions that you have installed:
Bookmarks,
Githu Light Theme,
GitLens,
SVG Viewer.
I try disable all extensions but it not work

this javascript code's tip is misleading, this is not inaccessible all time

by the way , in my pc this scipt is normal

in my mac's vscode:
[img](https://github.com/alexxxcs1/resource/blob/master/WechatIMG7.png?raw=true)
[img2](https://github.com/alexxxcs1/resource/blob/master/WechatIMG8.png?raw=true)

```
 function Test(params) {

    const foo = (()=>{

      try {

        return 1

      } catch (error) {

        return error

      }

    })();

    //this script is accessible!!!!!! but tips is not accessible

    const A = foo;

    const opacityDown = A;

    const MeanThisScriptNotAccessible = '?'

    //this script is accessible!!!!!! but tips is not accessible

}

```

**Comment by alexxxcs1** [comment]
@mjbvz > /needsMoreInfo
I'm done with the information

**Comment by mjbvz** [comment]
I confirmed this bug  when using TypeScript 3.8 for intellisense, however it seems to be fixed when using TypeScript 3.9 

You can upgrade your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next)
